drivers/sensor/tmp112/tmp112.c

```diff
@@ -215,6 +214,8 @@ int tmp112_init(const struct device *dev)
 	static const struct tmp112_config tmp112_cfg_##n = {				 \
 		.bus = DEVICE_DT_GET(DT_INST_BUS(n)),					 \
 		.i2c_slv_addr = DT_INST_REG_ADDR(n),					 \
+		.cr = DT_ENUM_IDX(DT_DRV_INST(n), sampling_rate),			 \
+		.extended_mode = DT_ENUM_IDX(DT_DRV_INST(n), full_scale),		 \
```

Comment:
an enum property being used as a boolean sounds weird, why not use a boolean in devicetree?

Reply:
That way the devicetree and `tmp112_attr_set(SENSOR_ATTR_FULL_SCALE)` are coherent, they accept the same values